Jenkinsfile

```diff
@@ -137,6 +137,23 @@ pipeline {
         }
       }
     }
+    stage('Building packages') {
+      agent { label 'docker && linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.PATH}:${env.WORKSPACE}/.local/bin"
+      }
+      steps {
+        deleteDir()
+        unstash 'source'
+        dir("${BASE_DIR}"){
+          sh script: 'pip3 install --user cibuildwheel', label: "Installing cibuildwheel"
+          sh script: 'mkdir wheelhouse', label: "creating wheelhouse"
+          sh script: 'cibuildwheel --platform linux --output-dir wheelhouse; ls -l wheelhouse'
+        }
+      }
+    }
   }
```

Comment:
```suggestion
      stage('Release') {
        agent { label 'linux && immutable' }
        options { skipDefaultCheckout() }
        environment {
          HOME = "${env.WORKSPACE}"
          PATH = "${env.PATH}:${env.WORKSPACE}/.local/bin"
        }
        when {
          beforeAgent true
          anyOf {
            tag "v\\d+\\.\\d+\\.\\d+*"
            expression { return params.Run_As_Master_Branch }
          }
        }
        steps {
          input(message: 'Should we release a new version?', ok: 'Yes, we should.')
          deleteDir()
          unstash 'source'
          unstash('packages')
          dir("${BASE_DIR}"){
            /** TODO code to release */
          }
        }
      }
    }
```

Reply:
this stage will be trigger when we create a tag like vNumber.Number.Number, it'd take the packages that we stash before and push them to some place

Reply:
Nice!

The code to push the release is relatively simple:

https://github.com/elastic/apm-agent-python/blob/b3b0ba42c260935b1d3c3a1db244c1ebc14d40ad/.travis.yml#L36-L39

(we'd also need to make the PyPI password available as environment variable)

Reply:
we can use getVaultSecret('secret') to grab the credentials something like this

```
  def jsonValue = getVaultSecret(secret: 'secret')
  wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [
    [var: 'TWINE_USER', password: jsonValue.data.user],
    [var: 'TWINE_PASSWORD', password: jsonValue.data.password],
    ]]) {
    withEnv([
      "TWINE_USER=${jsonValue.data.user}",
      "TWINE_PASSWORD=${jsonValue.data.password}"]) {
        sh(label: "Release packages", script: """
        python -m pip install twine 
        python setup.py sdist 
        python -m twine upload --skip-existing --repository-url https://upload.pypi.org/legacy/ dist/*.tar.gz 
        python -m twine upload --skip-existing --repository-url https://upload.pypi.org/legacy/ wheelhouse/*.whl
        """)
    }
  }
```

mind note: we need a `withSecretVault('secret'){ /** code block */ }` step to make things simpler